react-ui/src/components/SignupForm.test.js

```diff
@@ -0,0 +1,112 @@
+import React from "react";
+import { create } from "react-test-renderer";
+import { render, waitFor, cleanup, fireEvent, queryByAttribute, getByDisplayValue } from '@testing-library/react';
+import * as axios from 'axios';
+
+import SignupForm from './SignUpForm';
+import { BrowserRouter } from "react-router-dom";
+import { last } from "lodash";
+
+jest.mock('axios');
+
+
+describe('SignupForm component', () => {
+  afterEach(cleanup);
+
+  beforeEach(() => {
+    jest.resetModules(); // Clears any cache between tests.
+  });
+
+  test('first name input in signup form', () => {
+    //Arrange
+    const dom = render(
+      <BrowserRouter>
+        <SignupForm />
+      </BrowserRouter>);
+      const newValue = "Bob";
+
+    // Act
+    const getByName = queryByAttribute.bind(null, 'name');
+    const firstNameInput = getByName(dom.container, 'first_name');
+    fireEvent.change(firstNameInput, { target: { value: newValue } })
+
+    //Assert
+    expect(firstNameInput.value).toBe(newValue);
+
+    //console.log('dom', this.state);
+    //console.log('before', first_name.value);
+    //console.log('after', first_name.value);
+  });
+  
+  test('last name input in signup form', () => {
+    //Arrange
+    const dom = render(
+      <BrowserRouter>
+        <SignupForm />
+      </BrowserRouter>);
+      const newValue = "World";
+
+    // Act
+    const getByName = queryByAttribute.bind(null, 'name');
+    const lastNameInput = getByName(dom.container, 'last_name');
+    fireEvent.change(lastNameInput, { target: { value: newValue } })
+
+    //Assert
+    expect(lastNameInput.value).toBe(newValue);
+    // console.log('before', lastNameInput.value);
```

Comment:
Left in console.log "debugging" statements for now to assist with future development of frontend unit tests which are still to be setup for other components.